### PR TITLE
unix: Fail gracefully when calling uv_shutdown twice

### PR DESCRIPTION
**Pull request by Keno**
Before this, one of the requests was left as a ghost in the system. See https://github.com/JuliaLang/julia/issues/5793

### CODE REVIEW

**Comment by indutny** [comment]
Wow, thanks for catching it! May I ask you to write a test for it? (and also fix all issues mentioned by @Nodejs-Jenkins )

**Comment by Keno** [comment]
Updated.

File: test/test-shutdown-twice.c [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  ASSERT(0 == uv_ip4_addr("127.0.0.1", TEST_PORT, &addr));
+  loop = uv_default_loop();
+
+  r = uv_tcp_init(loop, &h);
+
+  r = uv_tcp_connect(&connect_req,
+                     &h,
+                     (const struct sockaddr*) &addr,
+                     connect_cb);
+  printf("%d\n", r);
```
Comment (saghul) [inline test/test-shutdown-twice.c:75]: Debug leftover?

**Comment by saghul** [comment]
LGTM. If you don't mind please remove the printf, otherwise I can remove it when merging.

**Comment by saghul** [comment]
Thanks @loladiro, landed in https://github.com/joyent/libuv/commit/a284b90c03735095763523a7c63f80140dc6a03a

**Comment by saghul** [comment]
For the record, I made a backport for 0.10 and landed it in https://github.com/joyent/libuv/commit/6e2021ca111a44449932792e0cb578e63705c2a2

**Pull request closed without merging**